substra/sdk/client.py

```diff
@@ -49,26 +56,158 @@ def wrapper(*args, **kwargs):
+    if env_var_name in os.environ:
+        return SettingValue(value=mapping[attribute_name](os.environ[env_var_name]), origin="environment_variable")
+
+    if attribute_name in client_config_dict:
+        return SettingValue(value=mapping[attribute_name](client_config_dict[attribute_name]), origin="config_file")
+
+    return SettingValue(value=None, origin="default")
+
+
+def get_client_configuration(
+    *,
+    config_file: Optional[pathlib.Path],
+    configuration_name: Optional[str],
+    code_values: Dict[str, Any],
+) -> Dict[str, SettingValue]:
+    """
+    Get the unified configuration values for a given client, out of three possible sources.
+    The order of precedence is: values defined by the user in the code, environment variables, values read from the
+    configuration file. If the attribute is not set, the value returned is None (and the origin is set to "default").
+
+    Args:
+        config_file (Path or None): Path to the yaml configuration file, if existing
+        configuration_name (str or None): Name of the configuration to use; it is used to determine which env var to use
+            and which sub-dictionary of the configuration file to consider; if None, only the values defined in the code
+            are considered.
+        code_values (dict): Value passed by the user directly in the code when instantiating the `Client`
+
+    Returns:
+     A dictionary associating each attribute with its `SettingValue` (containing the actual value and its origin)
+
+    """
+    if config_file and config_file.exists():
+        config_dict = yaml.safe_load(config_file.read_text())
+    else:
+        config_dict = {}
+
+    values_mapping = {
+        "url": str,
+        "token": str,
+        "username": str,
+        "password": str,
+        "retry_timeout": int,
+        "insecure": bool,
+        "backend_type": schemas.BackendType,
+    }
+    return {
+        attribute_name: _get_config_value(
+            attribute_name=attribute_name,
+            code_values=code_values,
+            configuration_name=configuration_name,
+            client_config_dict=config_dict.get(configuration_name, {}),
+            mapping=values_mapping,
+        )
+        for attribute_name in values_mapping
+    }
+
+
 class Client:
-    """Create a client
+    """Create a client.
+    Defaults to a subprocess client, suitable for development purpose.
+    Configuration can be passed in the code, through environment variables or through a configuration file.
 
     Args:
-        url (str, optional): URL of the Substra platform. Mandatory
-            to connect to a Substra platform. If no URL is given debug must be True and all
+        configuration_name (str, optional): Name of the configuration.
+            Used to load relevant environment variables and select the right dictionary in the configuration file.
+            If None, configuration values will only be read from the code, or default will be used.
```

Comment:
It could also be read from env variable ?
Maybe it could be easier to explain the order of priority once and for all in the intro text of the docsting, but not in the args.